src/Hamiltonians/Stoquastic.jl

```diff
@@ -0,0 +1,32 @@
+"""
+    Stoquastic(ham :> AbstractHamiltonian) :> AbstractHamiltonian
+A wrapper for an [`AbstractHamiltonian`](@ref) that replaces all off-diagonal matrix
+elements `v` by `-abs(v)`, thus making the new Hamiltonian *stoquastic*.
+
+A stoquastic Hamiltonian does not have a Monte Carlo sign problem. For a hermitian `ham`
+the smallest eigenvalue of `Stoquastic(ham)` is ≤ the smallest eigenvalue of `ham`.
+"""
+struct Stoquastic{T,H} <: AbstractHamiltonian{T}
+    hamiltonian::H
+end
+
+Stoquastic(h) = Stoquastic{eltype(h),typeof(h)}(h)
+
+starting_address(h::Stoquastic) = starting_address(h.hamiltonian)
+
+LOStructure(SH::Type{<:Stoquastic{<:Any,H}}) where {H} = _los_stoquastic(LOStructure(H), SH)
+
+_los_stoquastic(::IsHermitian, _) = IsHermitian()
+_los_stoquastic(::IsDiagonal, _) = IsDiagonal()
+_los_stoquastic(_, _) = AdjointUnknown()
+# Maybe implement AdjointKnown later?
+
```

Comment:
Why is this needed and why does it take two arguments?

To implement adjoints, I would go with
```suggestion
LOStructure(::Type{<:Stoquastic{<:Ant,H}}) where {H} = LOStructure(H)
Base.adjoint(h::Stoquastic) = Stoquastic(h.hamiltonian')
```
This circumvents the machinery that avoids computing adjoints if a Hamiltonian is Hermitian, but I don't think it's an issue as in those cases this will still return the exact same object. The only issue I see is it turning a non-Hermitian Hamiltonian into a Hermitian one - but I don't think that's a big deal either.

Reply:
Thanks! Great suggestion.